auditlog_security/models/auditlog_rule.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2021 Therp B.V.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import exceptions, models, fields, api, modules, _
+from odoo.addons.auditlog.models.rule import FIELDS_BLACKLIST
+
+
+class AuditlogRule(models.Model):
+    _inherit = "auditlog.rule"
+
+    auditlog_line_access_rule_ids = fields.One2many(
+        "auditlog.line.access.rule", "auditlog_rule_id", ondelete="cascade"
+    )
+    server_action_id = fields.Many2one('ir.actions.server', "Server Action",
+        ondelete="cascade"
+    )
+
+    @api.onchange("model_id")
+    def onchange_model_id(self):
+        # if model changes we must wipe out all field ids
+        self.auditlog_line_access_rule_ids.unlink()
+
+    @api.model
+    def _get_view_log_lines_action(self, model_id):
+        logs = self.env['auditlog.log'].sudo().search([
+            ('model_id', '=', model_id),
+            ('res_id', 'in', self.env.context.get('active_ids'))
+            ])
+        lines = self.env['auditlog.log.line'].sudo().search([
+            ('log_id', 'in', logs.ids) 
+            ])
```

Comment:
Two feedback points still here:

1. The first query should go, since we can now directly query on `model_id` on auditlog.log.line and we should do so for speed reasons
2. The second query should not have `sudo` because it should only return those that the user can actually access